documentation/latency-debugging-guide.md

```diff
@@ -0,0 +1,257 @@
+                        .build())
+                .addSpanProcessor(BatchSpanProcessor.builder(traceExporter).build())
+                .build())
+        .buildAndRegisterGlobal();
+String projectId = "my-project";
+String instanceId = "my-instance";
+String databaseId = "my-database";
+// Setting this to true instructs the JDBC driver to include the SQL statement with the traces.
+boolean enableExtendedTracing = true;
+
+try (Connection connection =
+   DriverManager.getConnection(
+       String.format(
+           "jdbc:cloudspanner:/projects/%s/instances/%s/databases/%s?enableExtendedTracing=%s",
+           projectId, instanceId, databaseId, enableExtendedTracing))) {
+ try (Statement statement = connection.createStatement()) {
+   try (ResultSet rs = statement.executeQuery("SELECT CURRENT_TIMESTAMP()")) {
+     while (rs.next()) {
+       System.out.printf(
+           "Connected to Cloud Spanner at [%s]%n", rs.getTimestamp(1).toString());
+     }
+   }
+ }
+}
+```
+
+### Example Using an OpenTelemetry instance in Properties
+Create and register a global OpenTelemetry object before creating a JDBC connection.
+See also the [Spring Data JDBC Sample](../samples/spring-data-jdbc) for an example for how to
+configure OpenTelemetry in combination with Spring Data.
+
+```java
+TraceConfiguration traceConfiguration = TraceConfiguration.builder().setProjectId("my-project").build();
+SpanExporter traceExporter = TraceExporter.createWithConfiguration(traceConfiguration);
+OpenTelemetry openTelemetry =
+    OpenTelemetrySdk.builder()
+        .setTracerProvider(
+            SdkTracerProvider.builder()
+                .setSampler(Sampler.traceIdRatioBased(0.05))
+                .setResource(
+                    Resource.builder()
+                        .put("service.name", "my-unique-service-name")
+                        .build())
+                .addSpanProcessor(BatchSpanProcessor.builder(traceExporter).build())
+                .build())
+        .build();
+String projectId = "my-project";
+String instanceId = "my-instance";
+String databaseId = "my-database";
+// Setting this to true instructs the JDBC driver to include the SQL statement with the traces.
+boolean enableExtendedTracing = true;
+
+Properties properties = new Properties();
+properties.put(JdbcDriver.OPEN_TELEMETRY_PROPERTY_KEY, openTelemetry);
+properties.put("enableExtendedTracing", String.valueOf(enableExtendedTracing));
+
+try (Connection connection =
+   DriverManager.getConnection(
+       String.format(
+           "jdbc:cloudspanner:/projects/%s/instances/%s/databases/%s",
+           projectId, instanceId, databaseId), properties)) {
+ try (Statement statement = connection.createStatement()) {
+   try (ResultSet rs = statement.executeQuery("SELECT CURRENT_TIMESTAMP()")) {
+     while (rs.next()) {
+       System.out.printf(
+           "Connected to Cloud Spanner at [%s]%n", rs.getTimestamp(1).toString());
+     }
+   }
+ }
+}
+```
+
+## Traces
```

Comment:
This only talks about JDBC traces, but in reality customer would see way more traces coming from spanner library.

Reply:
This text 'pretends' that all the traces come from the JDBC driver (although some of the text also mentions the client lib). I think that for a customer it is easier to read/understand if we mainly talk about JDBC here, as most users who use the JDBC driver are not using the client library. The fact that they are layered is not necessarily a known fact for everyone.
The traces that are described here also include the traces from the client library (e.g. everything in point 2 and 3 are actually collected by the client library)

Reply:
Actually, those are the ones that are described in point 2 below, but the text did not call out what they are named. I added that to the text to make it easier for a reader to make the link.